teamscale-maven-plugin/src/main/java/com/teamscale/maven/upload/CoverageUploadMojo.java

```diff
@@ -101,53 +144,85 @@ public void execute() throws MojoFailureException {
+					reportIntegrationGoalOutputFiles);
+			collectReportOutputDirectory(subProject, "report-aggregate", "jacoco-aggregate",
+					reportAggregateGoalOutputFiles);
 		}
 	}
 
 	/**
-	 * Collect the file locations in which JaCoCo is configured to save the XML coverage reports
-	 * @param project The project
-	 * @param reportGoal The JaCoCo report goal
-	 * @param jacocoDirectory The name of the directory, matching the JaCoCo goal
+	 * Collect the file locations in which JaCoCo is configured to save the XML
+	 * coverage reports
+	 *
+	 * @param project
+	 *            The project
+	 * @param reportGoal
+	 *            The JaCoCo report goal
+	 * @param jacocoDirectory
+	 *            The name of the directory, matching the JaCoCo goal
 	 * @see <a href="https://www.eclemma.org/jacoco/trunk/doc/maven.html">Goals</a>
 	 */
-	private void collectReportOutputDirectory(MavenProject project, String reportGoal, String jacocoDirectory, List<Path> reportOutputFiles) throws MojoFailureException {
+	private void collectReportOutputDirectory(MavenProject project, String reportGoal, String jacocoDirectory,
+			List<Path> reportOutputFiles) throws MojoFailureException {
 		Path defaultOutputDirectory = Paths.get(project.getReporting().getOutputDirectory());
 		// If a Dom is null it means the execution goal uses default parameters which work correctly
-		Xpp3Dom reportConfigurationDom = getJacocoGoalExecutionConfiguration(project,reportGoal);
+		Xpp3Dom reportConfigurationDom = getJacocoGoalExecutionConfiguration(project, reportGoal);
 		String errorMessage = "Skipping upload for %s as %s is not configured to produce XML reports for goal %s. See https://www.jacoco.org/jacoco/trunk/doc/report-mojo.html#formats";
 		if (!validateReportFormat(reportConfigurationDom)) {
-			throw new MojoFailureException(String.format(errorMessage, project.getName(), JACOCO_PLUGIN_NAME, jacocoDirectory));
+			throw new MojoFailureException(
+					String.format(errorMessage, project.getName(), JACOCO_PLUGIN_NAME, jacocoDirectory));
 		}
-		Path resolvedOutputDir = getCustomOutputDirectory(reportConfigurationDom).orElse(defaultOutputDirectory.resolve(jacocoDirectory).resolve("jacoco.xml"));
-		getLog().debug(String.format("Adding possible report location: %s", resolvedOutputDir));
-		reportOutputFiles.add(resolvedOutputDir);
+		Path resolvedCoverageFile = getCustomOutputDirectory(reportConfigurationDom).orElse(defaultOutputDirectory)
+				.resolve(jacocoDirectory).resolve("jacoco.xml");
+		getLog().debug(String.format("Adding possible report location: %s", resolvedCoverageFile));
+		reportOutputFiles.add(resolvedCoverageFile);
 	}
 
 	private void uploadCoverageReports() throws IOException {
-		uploadCoverage(reportGoalOutputFiles, unitTestPartition);
-		uploadCoverage(reportIntegrationGoalOutputFiles, integrationTestPartition);
-		uploadCoverage(reportAggregateGoalOutputFiles, aggregatedTestPartition);
+		if (uploadTestwiseCoverage) {
```

Comment:
Why don't we just upload everything all the time as we do with unit, integration and aggregated test coverage? Do you have some specific use case in mind?

Reply:
I thought that it seems unlikely that someone uploads testwise coverage while also wanting to upload raw jacoco coverage. Can change it so that everything is uploaded all the time. WDYT?